File: 2017/Python/day_2.py
def parse_input():
    full_split = []
    with open("Inputs/day_2_input.txt") as input_file:
        for line in input_file:
            splitter = line.split("\t")

            # Remove newline from the splitter
            splitter[len(splitter) - 1] = splitter[len(splitter) - 1].replace("\n", "")

            full_split.append(splitter)

    return full_split


def part_1():
    sumr = 0
    nums = parse_input()

    for i in range(len(nums)):
        nums_ss = nums[i]
        nums_ss = [int(string) for string in nums_ss]
        nums_ss.sort()

        sumr += nums_ss[len(nums_ss) - 1] - nums_ss[0]

    print("The checksum for Part 1 is: " + str(sumr))


def part_2():
    sumr = 0
    nums = parse_input()

    for i in range(len(nums)):
        nums_ss = nums[i]
        nums_ss = [int(string) for string in nums_ss]

        breaker = 0

        for j in nums_ss:
            for k in nums_ss:
                if j != k and j % k == 0:
                    sumr += j / k
                    breaker = 1
                    break
            if breaker == 1:
                break

    print("The checksum for Part 2 is: " + str(int(sumr)))


if __name__ == "__main__":
    part_1()
    part_2()

File: 2017/Rust/aoc_2017/src/day_2.rs
use std::fs;

pub fn day_2_main() {
    //Ingest data
    let file_name = "inputs/day_2_input.txt";
    let contents = fs::read_to_string(file_name).expect("Failed to read File");

    //Split by each line
    let parts: Vec<&str> = contents.split("\n").collect();

    println!("The answer the Part 1 is: {}", part_1(parts.clone()));
    println!("The answer the Part 2 is: {}", part_2(parts.clone()));
}

fn part_1(parts: Vec<&str>) -> i32 {
    let mut sumr = 0;

    for i in 0..parts.len() - 1 {
        let nums_str: Vec<&str> = parts[i].split("\t").collect();
        let mut nums: Vec<i32> = Vec::new();

        for j in 0..nums_str.len() {
            if let Ok(_) = nums_str[j].parse::<i32>() {
                nums.push(nums_str[j].parse::<i32>().unwrap());
            }
        }

        nums.sort();

        sumr += nums[nums.len() - 1] - nums[0];
    }

    return sumr;
}

fn part_2(parts: Vec<&str>) -> i32 {
    let mut sumr = 0;

    for i in 0..parts.len() - 1 {
        let nums_str: Vec<&str> = parts[i].split("\t").collect();
        let mut nums: Vec<i32> = Vec::new();

        for j in 0..nums_str.len() {
            if let Ok(_) = nums_str[j].parse::<i32>() {
                nums.push(nums_str[j].parse::<i32>().unwrap());
            }
        }

        'outer: for j in 0..nums.len() {
            for k in 0..nums.len() {
                if j != k && nums[j] % nums[k] == 0 {
                    sumr += nums[j] / nums[k];
                    break 'outer;
                }
            }
        }
    }

    return sumr;
}

